### PR TITLE
API diff between .NET 9 Preview 5 and .NET 9 Preview 6

### PR DESCRIPTION
**Pull request by carlossanlop**
Repo area owners:

- ASP.NET - @dotnet/aspnet-api-review 
- WinForms - @merriemcgaw @dreddy-work @JeremyKuhne
- WPF - @singhashish-wpf and @pchaurasia14
- Libraries - @artl93 @jeffschwMSFT @jeffhandley @karelz @ericstj @stephentoub @SamMonoRT 

Libraries area owners:
- System @dotnet/area-system-runtime 
- System.Buffers @dotnet/area-system-buffers 
- System.Collections @dotnet/area-system-collections 
- System.Diagnostics @dotnet/area-system-diagnostics @dotnet/area-system-diagnostics-metric 
- System.Drawing @dotnet/area-system-drawing 
- System.Formats.Nrbf @dotnet/area-system-formats-nrbf 
- System.IO @dotnet/area-system-io 
- System.Net @dotnet/ncl 
- System.Numerics @dotnet/area-system-numerics 
- System.Reflection @dotnet/area-system-reflection-emit 
- System.Runtime.InteropServices @dotnet/interop-contrib
- System.Runtime.Intrinsics @dotnet/area-system-runtime-intrinsics 
- System.Security @dotnet/area-system-security 
- System.Text.Json @dotnet/area-system-text-json 
- System.Text.RegularExpressions @dotnet/area-system-text-regularexpressions 

**Known api-diff tool issues**:

If yo usee any of these, please provide a GitHub suggestion in this PR to correct it: 

- The AsmDiff tool sometimes skips adding `{` or `{}` [arcade#10981](https://github.com/dotnet/arcade/issues/10981)(https://github.com/dotnet/arcade/issues/13814) 
- RequiresPreviewAttribute not getting auto added [arcade#14498](https://github.com/dotnet/arcade/issues/14498)
- Changes to `protected internal` visibility should stop showing. https://github.com/dotnet/arcade/issues/14579
- The AsmDiff tool shows APIs with decorating attribute changes as removed and re-added, instead of showing only the API. Please ignore this, it might be confusing but technically it's not incorrect: [arcade#13814]

### CODE REVIEW

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Diagnostics.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+-        public Activity StartActivity(string name, ActivityKind kind, ActivityContext parentContext, IEnumerable<KeyValuePair<string, object?>>? tags = null, IEnumerable<ActivityLink> links = null, DateTimeOffset startTime = default(DateTimeOffset));
++        public Activity? StartActivity(string name, ActivityKind kind, ActivityContext parentContext, IEnumerable<KeyValuePair<string, object?>>? tags = null, IEnumerable<ActivityLink>? links = null, DateTimeOffset startTime = default(DateTimeOffset));
+-        public Activity StartActivity(string name, ActivityKind kind, string parentId, IEnumerable<KeyValuePair<string, object?>>? tags = null, IEnumerable<ActivityLink> links = null, DateTimeOffset startTime = default(DateTimeOffset));
++        public Activity? StartActivity(string name, ActivityKind kind, string? parentId, IEnumerable<KeyValuePair<string, object?>>? tags = null, IEnumerable<ActivityLink>? links = null, DateTimeOffset startTime = default(DateTimeOffset));
```
Comment (tarekgh) [inline release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Diagnostics.md:26]: Nothing change in the nullability. I expect not seeing this diff.

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Numerics.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# System.Numerics
+
+``` diff
+ namespace System.Numerics {
```
Comment (tannergooding) [inline release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Numerics.md:4]: System.Numerics changes LGTM

CC. @dotnet/area-system-numerics

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Runtime.Intrinsics.Arm.md [hunk 3]
```diff
@@ -0,0 +1,1058 @@
...
++            public unsafe static void StoreVectorAndZip(ushort* address, (Vector128<ushort> Value1, Vector128<ushort> Value2) value);
++            public unsafe static void StoreVectorAndZip(uint* address, (Vector128<uint> Value1, Vector128<uint> Value2, Vector128<uint> Value3, Vector128<uint> Value4) value);
++            public unsafe static void StoreVectorAndZip(uint* address, (Vector128<uint> Value1, Vector128<uint> Value2, Vector128<uint> Value3) value);
++            public unsafe static void StoreVectorAndZip(uint* address, (Vector128<uint> Value1, Vector128<uint> Value2) value);
++            public unsafe static void StoreVectorAndZip(ulong* address, (Vector128<ulong> Value1, Vector128<ulong> Value2, Vector128<ulong> Value3, Vector128<ulong> Value4) value);
++            public unsafe static void StoreVectorAndZip(ulong* address, (Vector128<ulong> Value1, Vector128<ulong> Value2, Vector128<ulong> Value3) value);
++            public unsafe static void StoreVectorAndZip(ulong* address, (Vector128<ulong> Value1, Vector128<ulong> Value2) value);
+         }
+     }
+     public abstract class Sve : AdvSimd {
```
Comment (tannergooding) [inline release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Runtime.Intrinsics.Arm.md:621]: It's not new in this preview, but I want to explicitly call out that `Sve` is marked `Experimental` and will ship that way in .NET 9 (as an FYI to anyone who looks at these release notes)

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Runtime.Intrinsics.Arm.md [hunk 4]
```diff
@@ -0,0 +1,1058 @@
+# System.Runtime.Intrinsics.Arm
```
Comment (tannergooding) [inline release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Runtime.Intrinsics.Arm.md:1]: System.Runtime.Intrinsics.Arm changes LGTM.

CC. @kunalspathak, @dotnet/area-system-runtime-intrinsics 

Notably the name changes in AdvSimd were for new APIs introduced this release, API review opted for a different format to make things overall more consistent with `Vector<T>`

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Diagnostics.md (hunk 1 above)
Comment (tarekgh) [reply release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Diagnostics.md:26]: Other than that, Diagnostics.* LGTM!

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Runtime.Intrinsics.X86.md [hunk 5]
```diff
@@ -0,0 +1,545 @@
+# System.Runtime.Intrinsics.X86
+
+``` diff
+ namespace System.Runtime.Intrinsics.X86 {
++    public abstract class Avx10v1 : Avx2 {
```
Comment (tannergooding) [inline release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Runtime.Intrinsics.X86.md:5]: System.Runtime.Intrinsics.X86 LGTM

CC. @dotnet/avx512-contrib, @dotnet/area-system-runtime-intrinsics

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Runtime.Intrinsics.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# System.Runtime.Intrinsics
+
+``` diff
+ namespace System.Runtime.Intrinsics {
```
Comment (tannergooding) [inline release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Runtime.Intrinsics.md:4]: System.Runtime.Intrinsics LGTM.

CC. @dotnet/area-system-runtime-intrinsics

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# System
```
Comment (tannergooding) [inline release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.md:1]: Changes to System related to the System.Runtime area LGTM

CC. @dotnet/area-system-runtime

**Review by tannergooding — APPROVED** [review]

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Text.Json.Serialization.Metadata.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+     }
+     public sealed class JsonPropertyInfoValues<T> {
++        public Func<ICustomAttributeProvider>? AttributeProviderFactory { get; set; }
+     }
+     public abstract class JsonTypeInfo {
++        public ICustomAttributeProvider ConstructorAttributeProvider { get; }
++        public Type ElementType { get; }
++        public Type KeyType { get; }
+-        public IJsonTypeInfoResolver? OriginatingResolver { get; set; }
++        public IJsonTypeInfoResolver OriginatingResolver { get; set; }
```
Comment (eiriktsarpalis) [inline release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Text.Json.Serialization.Metadata.md:34]: The updates in nullability are incorrect. Wonder how that happened?

**Comment by bartonjs** [comment]
System.Security(.Cryptography) LGTM

**Review by adamsitnik — APPROVED** [review]
System.Formats.Nrbf LGTM!

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Text.Json.Serialization.Metadata.md (hunk 8 above)
Comment (carlossanlop) [reply release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Text.Json.Serialization.Metadata.md:34]: Is everything else (green) correct? Should I just remove the red lines? Or should I remove the whole file?

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Diagnostics.md (hunk 1 above)
Comment (carlossanlop) [reply release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Diagnostics.md:26]: So I should remove the whole file? There are new types at top. Should they stay?

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Text.Json.Serialization.Metadata.md (hunk 8 above)
Comment (eiriktsarpalis) [reply release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Text.Json.Serialization.Metadata.md:34]: Everything else is correct. The nullability updates in the existing APIs are wrong.

File: release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Diagnostics.md (hunk 1 above)
Comment (tarekgh) [reply release-notes/9.0/preview/preview6/api-diff/Microsoft.NETCore.App/9.0-preview6_System.Diagnostics.md:26]: No, the rest of the changes in the file should stay. Only `ActivitySource` need to be fixed to include only the following changes:

image

**Pull request merged**